test/root/integration/octaneIntegrationTests.js

```diff
@@ -0,0 +1,168 @@
+          }
+
+          const gotDefects = await octane.get(defectEntityName)
+            .query(Query.field('id').inComparison(createdIds).build())
+            .limit(limit)
+            .offset(offset)
+            .fields('name', 'description')
+            .orderBy('name')
+            .execute()
+          assertAllCreatedDefectsWereFound(gotDefects)
+          assert.strictEqual(gotDefects.data.length, limit) // assert that the limit worked
+          const receivedIds = gotDefects.data.map(data => data.id)
+          for (let i = 0; i < gotDefects.data.length; i++) {
+            assert.strictEqual(gotDefects.data[i].name, expectedDefects[i].name)
+            assert.strictEqual(gotDefects.data[i].description, expectedDefects[i].description)
+          }
+          receivedIds.sort()
+          expectedIds.sort()
+          assertIdListsMatch(receivedIds, expectedIds)
+        })
+      })
+    })
+
+    describe('update operations', () => {
+      it('updates single defect without id', async () => {
+        const defectUpdate = { name: 'Update 0 No Id' }
+        const updatedDefect = await octane.update(defectEntityName, defectUpdate)
+          .at(createdSingleId).execute()
+        assert(updatedDefect.id, createdSingleId)
+        const gotUpdatedDefect = await octane.get(defectEntityName).at(updatedDefect.id).fields('name').execute()
+        assert(gotUpdatedDefect.name, defectUpdate.name)
+      })
+      it('updates single defect with id', async () => {
+        const defectUpdate = { id: createdSingleId, name: 'Update 0 With Id' }
+        const updateSingleDefect = await octane.update(defectEntityName, defectUpdate).execute()
+        assert(updateSingleDefect.id, createdSingleId)
+        const gotUpdatedDefect = await octane.get(defectEntityName).at(updateSingleDefect.id).fields('name').execute()
+        assert(gotUpdatedDefect.name, defectUpdate.name)
+      })
+      it('updates multiple defects', async () => {
+        const defectsToUpdate = []
+        createdIds.sort()
+        createdIds.map((id) => {
+          defectsToUpdate.push({
+            name: 'Update ' + Math.random(),
+            id: id
+          })
+        })
+        const updateDefects = await octane.updateBulk(defectEntityName, defectsToUpdate).execute()
+        const receivedIds = updateDefects.data.map(data => data.id)
+        assertIdListsMatch(receivedIds, createdIds)
+        const gotUpdatedDefect = await octane.get(defectEntityName).fields('name')
+          .query(Query.field('id').inComparison(receivedIds).build())
+          .orderBy('id')
+          .execute()
+        for (let i = 0; i < receivedIds.length; i++) { assert(gotUpdatedDefect.data[i].name, defectsToUpdate[i].name) }
+      })
+    })
+
+    describe('delete operations', () => {
+      it('deletes single defects', async () => {
+        const deleteSingleDefect = await octane.delete(defectEntityName).at(createdSingleId).execute()
```

Comment:
I have this error when running this method: "Timeout of 2000ms exceeded. For async tests and hooks, ensure "done()" is called; if returning a Promise, ensure it resolves. ". I think this is because I didn't run all of the tests. Usually this would not need an authentication, but if I run it alone, it needs to authenticate.